File: 프로그래머스/lv2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    
    int total = progresses.size();
    int day = 0;
    int release = 0;
    for(int i = 0; i < total; i++){
        int cur_progress = progresses[i] + (speeds[i] * day);
        if (cur_progress >= 100) {
            release++;
        }
        else{
            if(release != 0) {
                answer.push_back(release);
                release = 0;
            }
            int need_day = (100 - cur_progress) / speeds[i];
            if((100 - cur_progress) % speeds[i] != 0) need_day++;
            day += need_day;
            release++;
        }
    }
    answer.push_back(release);
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    
    day = 0
    cnt = 0
    for progress, speed in zip(progresses, speeds):
        if progress + speed * day >= 100:
            cnt += 1
        else:
            if cnt > 0:
                answer.append(cnt)
                cnt = 0
            left_progress = 100 - (progress + speed * day)
            add_day = math.ceil(left_progress / speed)
            day += add_day
            cnt += 1
    
    answer.append(cnt)
    
    return answer